website/versioned_docs/version-v1.3_alpha/troubleshooting.md

```diff
@@ -49,3 +49,105 @@ You can generate the missing `.dat` files using the following command:
 ```bash
 pnpm build:circuits-c
 ```
+
+## contracts: `prove` command failure
+
+### Case `Commitment mismatch`
+
+If your log looks like following, that's because you have already run the `prove` command before. You could either do
+
+1. update MACI up-to-date, and try again
```

Comment:
might be worth explaining the difference between running with the cli or with contract tasks for more context (e.g. that contract tasks run everything in 1)
Also that if you run genProof once, the next time you run it the commitments will be different due to the salts being random

Reply:
> do you mean to ask the users to check if they are running on the latest version of the code?

right, since I didn't encounter the `commitment mismatch` bug after I updated to the latest dev branch (I should find out what make this difference, I think that might be because of some await update?)

> might be worth explaining the difference between running with the cli or with contract tasks for more context

Do you think I should add it in the troubleshooting section? or should put it elsewhere?

> Also that if you run genProof once, the next time you run it the commitments will be different due to the salts being random

Yes, and I think that's why the displayed error log is `commitment mismatch`. Let me add the description.

Reply:
> do you mean to ask the users to check if they are running on the latest version of the code?

I tried old commits but failed to reproduce this bug 🤔
Remove the first instruction.

Reply:
> right, since I didn't encounter the commitment mismatch bug after I updated to the latest dev branch (I should find out what make this difference, I think that might be because of some await update?)

did you re deploy from scratch that time or just re run prove?

> Do you think I should add it in the troubleshooting section? or should put it elsewhere?

maybe in another doc, but reference it here so users can read on it

Reply:
> did you re deploy from scratch that time or just re run prove?
redeploy all in my localhost

> maybe in another doc, but reference it here so users can read on it
sure, which section do you think is proper to put this content? (I think would mainly introduce the difference in how user call the function and how they define the configs/parameters?)